src/comm/UdpIODevice.cc

```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+ *
+ *   (c) 2009-2018 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+ *
+ * QGroundControl is licensed according to the terms in the file
+ * COPYING.md in the root of the source code directory.
+ *
+ ****************************************************************************/
+
+#include "UdpIODevice.h"
+#include <algorithm>
+
+UdpIODevice::UdpIODevice(QObject *parent) : QUdpSocket(parent)
+{
+    _buffer = QByteArray();
+}
+
+bool UdpIODevice::canReadLine() const
+{
+    return _buffer.indexOf('\n') > -1 || hasPendingDatagrams();
+}
+
+qint64 UdpIODevice::readLineData(char *data, qint64 maxSize)
```

Comment:
Take a look in `QTextStream::readLine`

Reply:
I took a look.
While it can work on a ByteArray, It wasn't happy when it got resized.
without resizing the ByteArray, it would just store all the data in memory, instead of freeing what got consumed.